examples/error_handling.rs

```diff
@@ -0,0 +1,26 @@
+use tide::utils::After;
+use tide::{Request, Response, Result, StatusCode};
+
+#[async_std::main]
+async fn main() -> Result<()> {
+    tide::log::start();
+    let mut app = tide::new();
+
+    app.middleware(After(|mut res: Response| async {
+        if let Some(err) = res.downcast_error::<tide::http::Error>() {
+            let msg = err.to_string().to_owned();
+            res.set_status(StatusCode::ImATeapot);
+            res.set_body(format!("Teapot Status: {}", msg));
+        }
+        Ok(res)
+    }));
+
+    app.at("/").get(|_req: Request<_>| async {
+        let error = tide::http::Error::from_str(StatusCode::BadRequest, "Uh oh");
```

Comment:
I think using a ? here would be more representative of how people would actually use this, and the purpose of an example is to show real usage. Maybe use a route with a param and try to convert it to a number?